crd/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+mod error;
+
+pub use crate::error::CrdError;
+use derivative::Derivative;
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use stackable_operator::Crd;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::str::FromStr;
+
+// TODO: We need to validate the name of the cluster because it is used in pod and configmap names, it can't bee too long
+// This probably also means we shouldn't use the node_names in the pod_name...
+
+#[derive(Clone, CustomResource, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[kube(
+    group = "spark.stackable.de",
+    version = "v1",
+    kind = "SparkCluster",
+    shortname = "sc",
+    namespaced
+)]
+#[kube(status = "SparkClusterStatus")]
+pub struct SparkClusterSpec {
+    pub master: SparkNode,
+    pub worker: SparkNode,
+    pub history_server: Option<SparkNode>,
+    pub image: String,
+    pub secret: Option<String>,
```

Comment:
Not used at the moment, is it?

Reply:
No but will be one of the first (re)implemented features i think :)